### DISCUSSION TITLE
Problem with local development. Docker error on supabase diff command

### DISCUSSION THREAD

**Discussion by DuncanLHS** [supabase/supabase#14112] [Questions] [answered]
I'm getting a problem crop up when running `supabase diff`. I'm getting a docker error:
```
Ports are not available: exposing port TCP 0.0.0.0:54320 -> 0.0.0.0:0: listen tcp 0.0.0.0:54320: bind: An attempt was made to access a socket in a way forbidden by its access permissions.
```
I'm fairly new to docker so might be missing something but it appears that when running `diff`, supabase fires up another docker postgres image on 54320:5432 with a randomised name and the error indicates there might be a conflict somewhere with the port being already in use.

I've run a windows terminal command to check ports in use, here's the output:
```
PS C:\Users\****> Get-Process -Id (Get-NetTCPConnection -LocalPort 54320).OwningProcess
Get-NetTCPConnection : No MSFT_NetTCPConnection objects found with property 'LocalPort' equal to '54320'.  Verify the
value of the property and retry.
At line:1 char:18
+ Get-Process -Id (Get-NetTCPConnection -LocalPort 54320).OwningProcess
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (54320:UInt16) [Get-NetTCPConnection], CimJobException
    + FullyQualifiedErrorId : CmdletizationQuery_NotFound_LocalPort,Get-NetTCPConnection
```

It correctly picks up the other ports in use by supabase but errors on this one suggesting no process is using 54320?

What I'm also confused over is the main postgres db using 54322:5432 and the newly started image using 54320:5432, I don't know enough about ports to know whether them both having the same remote port is part of the problem?

Unfortunately, the port of the second image (fired up by `diff`) appears to be the only one not configurable in config.toml. I've tried completely rebuilding the docker setup only to get the same error.

**Comment by DuncanLHS** [comment, marked as answer]
Solved my own problem.

tl:dr I ran the following in PowerShell and rebooted.
```
netsh int ipv4 set dynamic tcp start=57343 num=8192
netsh int ipv6 set dynamic tcp start=57343 num=8192
```

Hyper-V reserves a large number of ports at random within a range, if you're unlucky after a reboot it will reserve ports that will conflict with other apps.

This is a due to a known issue with docker for windows and a conflict with Hyper-V. https://github.com/docker/for-win/issues/3171
There are some dodgy solutions suggested in the issue there but an article that describes the problem and solution well is here https://medium.com/@sevenall/completely-solve-the-problem-of-docker-containers-not-starting-or-running-on-windows-10-due-to-port-57f16ed6143.

**Reply by kjellski** [reply]
Great find and an easy solution! 👍 🤓
